experimental/dds/tree2/src/core/tree/pathTree.ts

```diff
@@ -142,3 +143,54 @@ export function compareFieldUpPaths(a: FieldUpPath, b: FieldUpPath): boolean {
 	}
 	return compareUpPaths(a.parent, b.parent);
 }
+
+/**
+ * Status of the tree that a particular node in {@link EditableTree} and {@link UntypedTree} belongs to.
```

Comment:
This is being defined in terms of something it is not allowed to depend on. If moving this to core, you will need to come up with a new way to define this (perhaps in terms of anchors?)

Also, I don't there there is much in core that treats the root field key as special like there is in higher levels, so InDocument vs Deleted doesn't really make a lot of sense at this level.

Moving getDetachedFieldContainingPath to core seems perfectly fine, but moving the rest will need some reworking on documentation, and since these docs are app dev facing, thats not great.